wire_protocol/src/client_connection.rs

```diff
@@ -117,6 +117,7 @@ impl ClientConnection for ClientConnectionImpl<'_> {
 }
 
 pub async fn read_wirecommand(connection: &mut dyn Connection) -> Result<Replies, ClientConnectionError> {
+    connection.set_validity(false);
```

Comment:
Right now we have two concepts with very similar names. I don't know if they can be combined. 
See line 108 above. It is calling `connection.invalidate()` which is distinct from `connection.set_validity(false)` if these can't be made into one thing, perhaps one or both should be renamed.

Reply:
thanks Tom, good point!
I have removed the old logic as the new logic is doing the same job.